### PR TITLE
Add hardware support specification

### PR DESCRIPTION
**Pull request by jlaako**
Add a document listing supported validated and tested hardware and
potential known pending hardware support items.

### CODE REVIEW

**Comment by jlaako** [comment]
@skuusela can you comment if the USB ethernet is used in CI and if it has different hardware than listed here?

**Comment by skuusela** [comment]
@jlaako The way DUTs are connected in CI looks like this: https://github.com/01org/DAFT#1-overview.
The BeagleBone Black is emulating USB ethernet so not sure how to classify that hardware.

File: doc/hardware.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+ - Serial console
+ - USB 3.0 Type-A
+ - GPU with HDMI output
+ - Audio output over HDMI
+ - WiFi/BT adapter
+ - eMMC
+ - microSD
+
+Following on-board devices/interfaces has not been tested.
+
```
Comment (msivosuo) [inline doc/hardware.rst]: have not been tested

File: doc/hardware.rst [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ - USB 3.1 Type-C
+
+Peripherals
+===========
+
+USB Mass Storage
+----------------
+
+Devices compliant with USB Mass Storage device class has been tested to work.
```
Comment (msivosuo) [inline doc/hardware.rst]: have been tested to work

File: doc/hardware.rst [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+USB Communication
+-----------------
+
+Devices compliant with USB Communication Device Class (CDC) should work, but
+has not been tested.
```
Comment (msivosuo) [inline doc/hardware.rst]: have not been tested

File: doc/hardware.rst [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+USB Audio
+---------
+
+Devices compliant with USB Audio Class 1.0 and 2.0 should work, but has not
+been tested.
```
Comment (msivosuo) [inline doc/hardware.rst]: have been tested

**Comment by jlaako** [comment]
@skuusela OK, thanks, which driver at the client side talks to that interface? I assume it ends up as some class compliant device.

**Comment by skuusela** [comment]
@jlaako The driver seems to be cdc_ether.

**Comment by jlaako** [comment]
@skuusela ok, then we have at least partially tested USB CDC support so I can change that part in the doc from untested to tested.

**Comment by jlaako** [comment]
@mythi I think this initial revision should be good to go now

File: doc/hardware.rst [hunk 5]
```diff
@@ -0,0 +1,112 @@
+IoT Reference OS Kit for Intel(r) Architecture Hardware Support
+###############################################################
+
+IoT Reference OS Kit supports a selection of hardware that have been tested to
+work to a limited extent, specified here. IoT Reference OS Kit may also work
+on other Intel(r) Architecture hardware, but it have not been tested.
```
Comment (msivosuo) [inline doc/hardware.rst]: has not been tested

Comment (msivosuo) [inline doc/hardware.rst]: or validated?

**Comment by mythi** [comment]
@jlaako thanks. looks good

**Pull request merged**